frontends/concrete-python/concrete/fhe/mlir/utils.py

```diff
@@ -76,6 +76,68 @@ def flood_replace_none_values(table: list):
     assert_that(all(value is not None for value in table))
 
 
+def construct_table_multivariate(node: Node, preds: List[Node]) -> List[Any]:
+    """
+    Construct the lookup table for a multivariate node.
+
+    Args:
+        node (Node):
+            Multivariate node to construct the table for
+
+        preds (List[Node]):
+            ordered predecessors to `node`
+
+    Returns:
+        List[Any]:
+            lookup table corresponding to `node` and its input value
+    """
+
+    assert all(
+        description.is_encrypted and isinstance(description.dtype, Integer)
+        for description in node.inputs
+    )
+
+    packing_bit_width = sum(pred.properties["original_bit_width"] for pred in preds)
+    packed_values = range(0, 2**packing_bit_width)
+
+    np.seterr(divide="ignore")
+
+    table: List[Optional[Union[int, np.bool_, np.integer, np.floating, np.ndarray]]] = []
+    for packed_value in packed_values:
+        inputs = []
+
+        shift = 0
+        for description, pred in zip(node.inputs, preds):
+            assert isinstance(description.dtype, Integer)
+
+            bit_width = pred.properties["original_bit_width"]
+            is_signed = description.dtype.is_signed
+
+            value = (packed_value >> shift) & ((2**bit_width) - 1)
+            shift += bit_width
+
+            if is_signed:
+                value -= 2 ** (bit_width - 1)
+
+            inputs.append(np.ones(description.shape, dtype=np.int64) * value)
+
+        try:
+            evaluation = node(*inputs)
+            table.append(
+                evaluation if evaluation.min() != evaluation.max() else int(evaluation.min())
```

Comment:
can add a comment on why evaluation is not suitable directly

Reply:
It's to reduce memory usage. If the result consists of a single value, why should we store the whole tensor for it.

Reply:
either a comment or a specific naming would make that more obvious
compressed_tensor = ...
or
# compress constant tensor to 1 element

Reply:
Done.